File: Exercise1/React_Client_Project/clientsidehmo/src/models/member.ts
import { CoronaVaccine } from "./coronaVaccine";

export interface Member {
    id: number;
    first_name: string;
    last_name: string;
    social_number: string;
    date_of_birth: string;
    city: string;
    street: string;  
    house_number: string; 
    phone: string;
    mobile_phone: string;
    corona_vaccines: CoronaVaccine[];
    is_sick: boolean;
    discovery_date: string | null;
    date_of_recovery: string | null;
    image: string | null;
  }
  

File: Exercise1/Python_Server_Project/models/member.py
from create_db import db
from models.corona_vaccine import CoronaVaccine
from datetime import datetime

class Member(db.Model):
    __tablename__='member'
    id = db.Column(db.Integer, primary_key=True)
    first_name = db.Column(db.String(50), nullable=False)
    last_name = db.Column(db.String(50), nullable=False)
    social_number = db.Column(db.String, unique=True, nullable=False)  
    date_of_birth = db.Column(db.String(10), nullable=False)  
    phone = db.Column(db.String(20))
    mobile_phone = db.Column(db.String(20))
    city = db.Column(db.String(50))  
    street = db.Column(db.String(50))  
    house_number = db.Column(db.String(20))  
    is_sick = db.Column(db.Boolean)  
    discovery_date = db.Column(db.String(10))  
    date_of_recovery = db.Column(db.String(10))  
    corona_vaccines = db.relationship('CoronaVaccine', backref='member', lazy=True)
    image = db.Column(db.String(100))

    def __repr__(self):
        return f'<Member {self.first_name} {self.last_name}>'
